pkg/probe/probe_builder.go

```diff
@@ -6,12 +6,15 @@ import (
 )
 
 type ProbeBuilder struct {
-	probeType          string
-	probeCategory      string
-	registrationClient TurboRegistrationClient
-	discoveryClientMap map[string]TurboDiscoveryClient
-	actionClient	   TurboActionExecutorClient
-	builderError       error
+	probeConf              *ProbeConfig
+	registrationClient     TurboRegistrationClient
+	discoveryClientMap     map[string]TurboDiscoveryClient
```

Comment:
`map[string]TurboDiscoveryClient`

   In our use cases,  there is always only one discoveryClient for a target. So can we get rid of the map?